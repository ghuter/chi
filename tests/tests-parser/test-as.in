Point :: struct {
	x: i32;
	y: i32;
}

f :: fun () {
	p :: Point {x: 0, y: 0};
	ptr :: (@p) as ^u64;

	x :: 10;
	y :: x as i8;
}
