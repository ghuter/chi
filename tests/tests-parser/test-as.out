SSTRUCT(Point, x : i32, y : i32)
SFUN(f, {SSEQ(const SDECL(p, ESTRUCT(Point { <x : ECSTI(0)> <y : ECSTI(0)> })) : ?, SSEQ(const SDECL(ptr, ECAST(EUNOP(AT, EMEM(p) : ?) : ? as ^u64) : ?, SSEQ(const SDECL(x, ECSTI(10)) : ?, const SDECL(y, ECAST(EMEM(x) : ? as i8) : ?)))}) : void
